test/utils/test_split.py

```diff
@@ -9,13 +11,26 @@ def test_generate_random_split():
     val_ratio = 0.1
     test_ratio = 0.1
 
-    split = generate_random_split(num_data, seed=42, train_ratio=train_ratio,
-                                  val_ratio=val_ratio)
-    assert (split == SPLIT_TO_NUM['train']).sum() == int(num_data *
-                                                         train_ratio)
-    assert (split == SPLIT_TO_NUM['val']).sum() == int(num_data * val_ratio)
-    assert (split == SPLIT_TO_NUM['test']).sum() == int(num_data * test_ratio)
+    split = generate_random_split(num_data, seed=42,
+                                  ratios=[train_ratio, val_ratio])
```

Comment:
why do we want this to be a list? I think the previous argument is clearer.

Reply:
This format is more aligned with split method in torch https://pytorch.org/docs/stable/_modules/torch/utils/data/dataset.html#random_split. And that to support split with only train and validation data with the new format, we are able to do `split(ratios=[0.8], ...)` while with the old format, we could only do `split(train_ratio=0.8, val_ratio=None, ...)` which is a bit weird, because this indicates a train/test split instead of a train/val split which is more common.

Reply:
Yeah but here we are actually assigning TRAIN to the first ratio, VAL to the second ratio, and so on. So I still think the previous one is better.

Reply:
torch's random split makes sense since it just splits the dataset. There is no train/val/test assignment happening under the hood.

Reply:
In the previous format we always assume we want to have a test set. So
1. it does not have the flexibility to doing something like (train: 0.3, val: 0.1, test: 0.2)
2. when we do train: 0.8, val: 0.0. Then the inferred test ratio is 0.2. Which is a bit weird because in this case only train and val (test) split is well-defined. i.e. we only intend to split the data into 2 parts instead of 3 parts. But we end up with an additional validation set that is completely empty

We could also keep the previous format but support the case where val ratio could be `None` which indicates there is no validation set instead of an empty validation set if you have strong opinion here.

Reply:
I understand you. but I still want features based on needs.
I don't find these logics useful for now, and having new interface would break other parts. For instance, this PR does not modify `generate_random_split` used in `data_frame_benchmark.py` and elsewhere.

Reply:
> For instance, this PR does not modify generate_random_split used in data_frame_benchmark.py and elsewhere

This is still a draft now. I kind of want to sync with you on this first before changing other parts of the code